Mathlib/Tactic/Widget/Calc.lean

```diff
@@ -124,19 +126,30 @@ def CalcPanel : Component CalcParams :=
   mk_rpc_widget% CalcPanel.rpc
 
 namespace Lean.Elab.Tactic
-open Meta
+
+open Lean Meta Tactic TryThis in
+/-- Create a `calc` proof. -/
+elab stx:"calc?" : tactic => withMainContext do
+  let some calcRange := (← getFileMap).rangeOfStx? stx | throwError "calc? failed"
+  let indent := calcRange.start.character + 2
+  let spc := String.replicate indent ' '
+  let goalType ← getMainTarget
+  unless (← Lean.Elab.Term.getCalcRelation? goalType).isSome do
+    throwError "Cannot start a calculation here: the goal{indentExpr goalType}\nis not a relation."
+  let goalFmt ← Meta.ppExpr (← getMainTarget)
+  let s := s!"calc\n{spc}{goalFmt} := by sorry"
+  addSuggestions (← getRef) #[.suggestion s] (header := "Create calc tactic:")
```

Comment:
I think it would be better to construct this from a `TSyntax` object than from a string; was there a reason you didn't do that?

Reply:
That is
```suggestion
  let s ← `(tactic| calc $(← Lean.PrettyPrinter.delab (← getMainTarget)) := by sorry)
  addSuggestions (← getRef) #[.suggestion s] (header := "Create calc tactic:")
```

Reply:
I specifically want to handle spaces here.

Reply:
According to my test, the `Syntax` version already handles the spaces correctly for you in all the ways that matter (everything but `guard_msgs`).

Reply:
In case it is helpful, my local test also *displays* the wrong spacing, but the action really inserts the correct indentation.

Reply:
> In case it is helpful, my local test also displays the wrong spacing,

I think it is worth reporting this upstream, but I don't think it is worth working around. In fact, the workaround here is actually broken if `goalFmt` contains a newline, but (I assume) works correctly with my suggestion.